packages/kit/src/runtime/server/respond.js

```diff
@@ -375,12 +380,28 @@ export async function respond(request, options, manifest, state) {
 					);
 				} else if (route.endpoint && (!route.page || is_endpoint_request(event))) {
 					response = await render_endpoint(event, await route.endpoint(), state);
-				} else if (route.page) {
+				} else if (route.page && page_methods.has(method)) {
+					console.log(page_methods.has(method));
 					response = await render_page(event, route.page, options, manifest, state, resolve_opts);
+				} else if (route.page && method === 'OPTIONS') {
+					// if no OPTIONS endpoint exists, fallback to handling it gracefully.
+					const allowed_methods = new Set(allowed_page_methods);
+
+					const node = await manifest._.nodes[route.page.leaf]();
+					if (node?.server?.actions) {
+						allowed_methods.add('POST');
+					}
+
+					response = new Response(null, {
```

Comment:
This does also send a 204 for cross origin OPTIONS requests. I think in that case we should play it safe and always do 405 - at least that's how I read the prior discussion.

Reply:
Ah, sorry. I must have gotten it mixed up again. I'll change it to send a 405 instead.
How about this implementation that always returns 405, and just hides the body if it's an `OPTIONS` request.
```js
				} else {
					// if the request method is not allowed by us for the page / endpoint.
					const mod = route.endpoint ? await route.endpoint() : {};

					let allowed_http_methods = new Set([...allowed_methods(mod)]);

					if (route.page) {
						allowed_http_methods = new Set([...allowed_http_methods, ...allowed_page_methods]);
						const node = await manifest._.nodes[route.page.leaf]();
						if (node?.server?.actions) {
							allowed_http_methods.add('POST');
						}
					}

					const body = method === 'OPTIONS' ? null : `${request.method} method not allowed`;
					return new Response(body, {
						status: 405,
						headers: {
							allow: Array.from(allowed_http_methods).join(', ')
						}
					});
				}
```

Reply:
Will this have unintended consequences? I have to admit I still don't fully understand the OPTIONS request in non-cross-origin environments.

Reply:
Turns out I was wrong - this implictly denies CORS requests because the required response headers aren't set, so we're good.